### ISSUE TITLE
Pretty print service failed

### ISSUE THREAD

**Issue by neofreko** [docker/for-mac#38]
### Expected behavior
Service information pretty-printed to console

### Actual behavior
Runtime error

```
➜  docker service inspect --pretty swarmbundle_web
ID:		de179gar9d0o7ltdybungplod
Name:		swarmbundle_web
Labels:
 - com.docker.stack.namespace=swarmbundle
Mode:		Replicated
 Replicas:	2
Placement:
UpdateConfig:
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4333ccb]

goroutine 1 [running]:
panic(0x47ee2a0, 0xc8200100e0)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/docker/docker/api/client/service.printService(0x5a44198, 0xc82007e008, 0xc8200122e0, 0x19, 0x13b, 0xecf2e2b14, 0x3104b62f, 0x4d29780, 0xecf2f27d5, 0x34878f30, ...)
	/go/src/github.com/docker/docker/api/client/service/inspect.go:121 +0x1d8b
github.com/docker/docker/api/client/service.printHumanFriendly(0x5a44198, 0xc82007e008, 0xc820333d00, 0x1, 0x2, 0xc82034acf0, 0x0, 0x0)
	/go/src/github.com/docker/docker/api/client/service/inspect.go:75 +0x19d
github.com/docker/docker/api/client/service.runInspect(0xc82009e510, 0xc820333d00, 0x1, 0x2, 0x0, 0x0, 0x1, 0x0, 0x0)
	/go/src/github.com/docker/docker/api/client/service/inspect.go:66 +0x239
github.com/docker/docker/api/client/service.newInspectCommand.func1(0xc820329440, 0xc820333d00, 0x1, 0x2, 0x0, 0x0)
	/go/src/github.com/docker/docker/api/client/service/inspect.go:40 +0xd9
github.com/spf13/cobra.(*Command).execute(0xc820329440, 0xc820333bc0, 0x2, 0x2, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/spf13/cobra/command.go:593 +0x6fe
github.com/spf13/cobra.(*Command).ExecuteC(0xc820071b00, 0xc820329440, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/spf13/cobra/command.go:683 +0x55c
github.com/spf13/cobra.(*Command).Execute(0xc820071b00, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/spf13/cobra/command.go:642 +0x2d
github.com/docker/docker/cli/cobraadaptor.CobraAdaptor.run(0xc820071b00, 0xc82009e510, 0x7fff5fbffaf7, 0x7, 0xc820082020, 0x3, 0x3, 0x0, 0x0)
	/go/src/github.com/docker/docker/cli/cobraadaptor/adaptor.go:118 +0x25f
github.com/docker/docker/cli/cobraadaptor.CobraAdaptor.Command.func1(0xc820082020, 0x3, 0x3, 0x0, 0x0)
	/go/src/github.com/docker/docker/cli/cobraadaptor/adaptor.go:126 +0x8c
github.com/docker/docker/cli.(*Cli).Run(0xc82034a660, 0xc820082010, 0x4, 0x4, 0x0, 0x0)
	/go/src/github.com/docker/docker/cli/cli.go:81 +0x34b
main.main()
	/go/src/github.com/docker/docker/cmd/docker/docker.go:72 +0x599
```

```
➜  docker service ls
ID            NAME               REPLICAS  IMAGE                                                                                         COMMAND
42wc6rtrcxay  swarmbundle_redis  2/2       redis@sha256:b50f15d427aea5b579f9bf972ab82ff8c1c47bffc0481b225c6a714095a9ec34
de179gar9d0o  swarmbundle_web    2/2       neofreko/flask-hello@sha256:817175eb246636d202ce2f85ae98ea1592cc22450abe5ff695dfb0762394d5af
```

```
➜  docker service inspect swarmbundle_web
[
    {
        "ID": "de179gar9d0o7ltdybungplod",
        "Version": {
            "Index": 315
        },
        "CreatedAt": "2016-07-30T04:59:00.822392367Z",
        "UpdatedAt": "2016-07-30T22:57:25.881299248Z",
        "Spec": {
            "Name": "swarmbundle_web",
            "Labels": {
                "com.docker.stack.namespace": "swarmbundle"
            },
            "TaskTemplate": {
                "ContainerSpec": {
                    "Image": "neofreko/flask-hello@sha256:817175eb246636d202ce2f85ae98ea1592cc22450abe5ff695dfb0762394d5af"
                }
            },
            "Mode": {
                "Replicated": {
                    "Replicas": 2
                }
            },
            "Networks": [
                {
                    "Target": "5vpyomhb6ievnk0i0o60gcnei",
                    "Aliases": [
                        "web"
                    ]
                }
            ],
            "EndpointSpec": {
                "Mode": "vip",
                "Ports": [
                    {
                        "Protocol": "tcp",
                        "TargetPort": 5000
                    }
                ]
            }
        },
        "Endpoint": {
            "Spec": {
                "Mode": "vip",
                "Ports": [
                    {
                        "Protocol": "tcp",
                        "TargetPort": 5000
                    }
                ]
            },
            "Ports": [
                {
                    "Protocol": "tcp",
                    "TargetPort": 5000,
                    "PublishedPort": 30000
                }
            ],
            "VirtualIPs": [
                {
                    "NetworkID": "6o4107cj2jx9tihgb0jyts6pj",
                    "Addr": "10.255.0.4/16"
                },
                {
                    "NetworkID": "5vpyomhb6ievnk0i0o60gcnei",
                    "Addr": "10.0.1.2/24"
                }
            ]
        },
        "UpdateStatus": {
            "StartedAt": "0001-01-01T00:00:00Z",
            "CompletedAt": "0001-01-01T00:00:00Z"
        }
    }
]
```

### Information

Diagnostic ID: B89DA7DC-71FF-437E-B968-4B16D3AE87B7
Docker for Mac: 1.12.0-beta21 (Build 10868)
macOS: Version 10.11.4 (Build 15E65)
[OK]     docker-cli
[OK]     app
[OK]     moby-syslog
[OK]     disk
[OK]     virtualization
[OK]     system
[OK]     menubar
[OK]     osxfs
[OK]     db
[OK]     slirp
[OK]     moby-console
[OK]     logs
[OK]     vmnetd
[OK]     env
[OK]     moby
[OK]     driver.amd64-linux

### Steps to reproduce

1. deploy service using bundle. Made from this compose file:
    
    ```
    version: '2'
    services:
    web:
        image: neofreko/flask-hello
        ports:
        - "5000"
        depends_on:
        - redis
    redis:
        image: redis
    ```
2. inspect deployed service

**Comment by dave-tucker** [comment]
This is now fixed. I'll update this ticket when the fix will be released.

**Comment by djs55** [comment]
Upstream fix for this was cherry-picked into docker 1.12.1. Docker 1.12.1 is now in the beta channel-- I confirmed by installing beta 24. This should therefore be fixed.
